test/pylib/s3_proxy.py

```diff
@@ -187,6 +192,8 @@ def process_request(self):
                 policy.error_count += 1
                 self.respond_with_error(reset_connection=policy.server_should_fail)
             else:
+                # Once the request is successfully processed, we remove the policy from the cache to make following request to the resource being illegible to fail
+                self.policies.remove(self.path)
```

Comment:
This doesn't seem to fit the patch subject. It's a separate fix, so separate patch, please

Reply:
Done